src/main/scala/com/microsoft/hyperspace/actions/CreateActionBase.scala

```diff
@@ -123,12 +126,33 @@ private[actions] abstract class CreateActionBase(dataManager: IndexDataManager)
         }
         // "path" key in options can incur multiple data read unexpectedly.
         val opts = options - "path"
-        Relation(
-          location.rootPaths.map(_.toString),
-          Hdfs(sourceDataProperties),
-          dataSchema.json,
-          fileFormatName,
-          opts)
+
+        // If user provided a globbing pattern, set rootPaths to this pattern. Else, use file index.
+        val rootPaths = opts.get(GLOBBING_PATTERN_KEY) match {
+          case Some(pattern) =>
+            // Validate if globbing pattern matches actual source paths.
+            // This logic is picked from the globbing logic at:
+            // https://github.com/apache/spark/blob/v2.4.4/sql/core/src/main/scala/org/apache/
+            // spark/sql/execution/datasources/DataSource.scala#L540
+            val globPaths = pattern.split(",").map(_.trim).map { path =>
+              val hdfsPath = new Path(path)
+              val fs = hdfsPath.getFileSystem(new Configuration)
+              val qualified = hdfsPath.makeQualified(fs.getUri, fs.getWorkingDirectory)
+              qualified.toString -> SparkHadoopUtil.get.globPathIfNecessary(fs, qualified)
+            }.toMap
+
+            val globPathValues = globPaths.values.flatten.toSet
```

Comment:
What if `spark.read.option(GLOBBING_PATTERN_KEY, "/*").parquet("/1/*")` ?
I think we could check it only at createIndex.. WDYT? @apoorvedave1

Reply:
Could you please test this once? In my experience the original df creation fails for the above example

Reply:
Yep looks like the case is also checked. Thanks for confirming!